### ISSUE TITLE
Removal of CompositeFilter

### ISSUE THREAD

**Issue by robobario** [kroxylicious/kroxylicious#624]
> How would that play with the idea of splitting up the multi tenant filter?

Personally I remain unconvinced that this is even a problem, never mind a problem that's worth trying to do something about. So you have a class which implements lots of interfaces and has lots of methods. Sure it's not _ideal_, but it's within the limits of the JVM. That class can instantiate your three subcomponents in its constructor and delegate to them if it wants. So the implementation is actually trivial.  

Will filters with that many interfaces be _common_? Probably not. 

tl;dr for me it just doesn't justify the extra complexity. The extra code complexity for filter authors just isn't that huge, but the extra cognitive load for them to grok how the more complex model it supposed to work just isn't worth it.

_Originally posted by @tombentley in https://github.com/kroxylicious/kroxylicious/issues/623#issuecomment-1730777706_

**Comment by gracegrimwood** [comment]
Triaged 1 Nov 2023 - CompositeFilter is deprecated, needs removal (not a priority).

**Comment by k-wall** [comment]
Done by #973.
